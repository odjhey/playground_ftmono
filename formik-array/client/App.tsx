import React from 'react'
import { View, TextInput } from 'react-native'
import { Button, Text } from 'react-native-elements'
import { Formik, getIn, FieldArray, Field } from 'formik'
import * as Yup from 'yup'

type item = {
  key: string
  name: string
  other?: string
}

const ItemSchema = Yup.object().shape({
  items: Yup.array().of(
    Yup.object().shape({
      key: Yup.string().required('Required'),
      name: Yup.string()
        .min(2, 'Too Short!')
        .max(50, 'Too Long!')
        .required('Required'),
      other: Yup.string(),
    }),
  ),
})

const data: Array<item> = [
  {
    key: '1',
    name: 'the names',
    other: 'lol',
  },
  {
    key: '2',
    name: 'another naem',
  },
]

export const MyReactNativeForm = (props: any) => {
  return (
    <View>
      <Formik
        initialValues={{ items: data }}
        validationSchema={ItemSchema}
        onSubmit={values => console.log(values)}>
        {({ handleChange, handleBlur, handleSubmit, values }) => (
          <View>
            <FieldArray
              name="items"
              render={fArrayBag => {
                return values.items.map((item, i) => (
                  <View
                    key={item.key}
                    style={{
                      elevation: 3,
                      padding: 2,
                      margin: 5,
                      borderColor: 'black',
                      borderWidth: 1,
                      borderStyle: 'dashed',
                    }}>
                    <TextInput
                      onChangeText={handleChange(`items[${i}].key`)}
                      onBlur={handleBlur(`items[${i}].key`)}
                      value={item.key}
                    />
                    <TextInput
                      onChangeText={handleChange(`items[${i}].name`)}
                      onBlur={handleBlur(`items[${i}].name`)}
                      value={item.name}
                    />
                    <ErrorMessage name={`items[${i}.name]`}></ErrorMessage>
                    <TextInput
                      onChangeText={handleChange(`items[${i}].other`)}
                      onBlur={handleBlur(`items[${i}].other`)}
                      value={item.other}
                    />
                  </View>
                ))
              }}></FieldArray>
            <Button onPress={handleSubmit} title="Submit" />
          </View>
        )}
      </Formik>
    </View>
  )
}

const ErrorMessage = ({ name }) => (
  <Field name={name}>
    {({ form }) => {
      const error = getIn(form.errors, name)
      const touch = getIn(form.touched, name)
      return touch && error ? (
        <Text style={{ color: 'red' }}>{error}</Text>
      ) : null
    }}
  </Field>
)

export default MyReactNativeForm
