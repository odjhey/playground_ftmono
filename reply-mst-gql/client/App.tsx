import React from 'react';
import {ScrollView, View, Text, TextInput} from 'react-native';
import {createHttpClient} from 'mst-gql';
import {observer} from 'mobx-react';
import {RootStore, StoreContext} from './src/models';

import {useQuery} from './src/models';

const rootStore = RootStore.create(undefined, {
  gqlHttpClient: createHttpClient('http://localhost:4000/graphql'),
});

// @ts-ignore.
global.store = rootStore;

const App = () => {
  return (
    <StoreContext.Provider value={rootStore}>
      <Main></Main>
    </StoreContext.Provider>
  );
};

const Main = observer(() => {
  const {loading, data, error} = useQuery(store => store.retrieveComments());

  if (loading) {
    return <Text>Loading</Text>;
  }

  if (error) {
    return <Text>error.messages</Text>;
  }

  return (
    <ScrollView>
      {data?.comments.map((comment, i) => (
        <Reply key={i} comment={comment}></Reply>
      ))}
      <View style={{padding: 2}}>
        <TextInput
          style={{
            marginBottom: 5,
            padding: 2,
            borderColor: '#0fafff',
            borderStyle: 'solid',
            borderWidth: 1,
          }}></TextInput>
        <TextInput
          style={{
            borderColor: '#ffafff',
            borderStyle: 'solid',
            borderWidth: 1,
          }}></TextInput>
      </View>
    </ScrollView>
  );
});

const Reply = props => {
  const {id, message, replyTo} = props.comment;
  return (
    <View
      style={{
        margin: 5,
        padding: 2,
        borderColor: '#999',
        borderStyle: 'solid',
        borderWidth: 1,
      }}>
      {replyTo ? (
        <View style={{backgroundColor: '#eee'}}>
          <Text>
            <Text style={{color: '#ABC'}}>{replyTo.id}</Text>
            <Text> </Text>
            <Text style={{color: '#888'}}>{replyTo.message}</Text>
          </Text>
        </View>
      ) : null}
      <Text>{id}</Text>
      <Text>{message}</Text>
    </View>
  );
};

export default App;
