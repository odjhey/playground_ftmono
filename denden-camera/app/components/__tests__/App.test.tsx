/**
 * @format
 */

import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import App from '../app/App';

describe('App', () => {
  it('App renders correctly', () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});
