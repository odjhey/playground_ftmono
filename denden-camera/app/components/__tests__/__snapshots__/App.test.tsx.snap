// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App renders correctly 1`] = `
<ContextProvider
  value={
    Object {
      "__queryCache": Object {},
      "books": Object {},
      "filteredBooks": Array [],
      "users": Object {},
    }
  }
>
  <Memo(wrappedComponent) />
</ContextProvider>
`;
